### PR TITLE
:sparkles: Add errors/unmatched rules in the rulesets

### PR DESCRIPTION
**Pull request by pranavgaikwad**

### CODE REVIEW

File: analyzer-lsp-main
Comment (fabianvf) [inline analyzer-lsp-main:1]: I think you committed this binary accidentally, we should remove it from this PR and probably add it to the .gitignore as well

**Review by fabianvf — CHANGES_REQUESTED** [review]

File: hubapi/violations.go [hunk 1]
```diff
@@ -5,13 +5,28 @@ import (
...
+	// Tags list of generated tags from the rules in this ruleset.
+	Tags []string `yaml:"tags,omitempty"`
+	// Violations is a map containing violations generated for the
+	// matched rules in this ruleset. Keys are rule IDs, values are
+	// their respective generated violations.
+	Violations map[string]Violation `yaml:"violations,omitempty"`
+	// Errors is a map containing errors generated during evaluation
+	// of rules in this ruleset. Keys are rule IDs, values are
+	// their respective generated errors.
+	Errors map[string]string `yaml:"errors,omitempty"`
```
[the remark this answers is not in the document]
Comment (pranavgaikwad) [reply hubapi/violations.go:27]: Not right now

**Review by shawn-hurley — APPROVED** [review]
I think this is an excellent first step I do think that we are going to need to update the API everywhere :)

**Comment by pranavgaikwad** [comment]
Following up on multiple errors thing in issue here https://github.com/konveyor/analyzer-lsp/issues/133

**Pull request merged**